src/Engine/Engine.cpp

```diff
@@ -1975,12 +1977,19 @@ void initLevelStrings(Blob &blob) {
 
     int offs = 0;
     while (offs < blob.size()) {
-        std::string str = &blob.string_view()[offs];
+        const char *nextNullTerm = (const char*)memchr(&blob.string_view()[offs], '\0', blob.size() - offs);
+        size_t stringSize = nextNullTerm ? (nextNullTerm - &blob.string_view()[offs]) : (blob.size() - offs);
+        std::string str = std::string(&blob.string_view()[offs], stringSize);
         if (str[0] == '"') {
-            str.erase(0, 1);
+            assert(str[str.length() - 1] == '"');
+            if (str.size() > 2) {
+                str = str.substr(1, str.size() - 2);
+            } else {
+                str = "";
+            }
```

Comment:
Why don't you just use `trimRemoveQuotes`? Or if you don't like the trim, then implement `removeQuotes` for `std::string` in `Utility/String`? It'll come handy later.

Also, I think asserting on input that you don't explicitly control (e.g. that might come from data created by modders) is generally a bad idea. Either throw or just make it work anyway.

Also that `str[0]` will crash on empty strings. Not sure if those are actually possible, but then again, we're talking about input you don't control.